packages/central-server/src/apiV2/utilities/constructNewRecordValidationRules.js

```diff
@@ -446,26 +446,45 @@ export const constructForSingle = (models, recordType) => {
         survey_id: [constructRecordExistsWithId(models.survey)],
         assignee_id: [constructIsEmptyOr(constructRecordExistsWithId(models.user))],
         due_date: [
-          (value, { status }) => {
-            if (status !== 'repeating' && !value) {
-              throw new Error('Due date is required for non-recurring tasks');
+          (value, { repeat_schedule: repeatSchedule }) => {
+            if (repeatSchedule) {
+              if (value) {
+                throw new Error('Non-recurring tasks must not have a due date');
+              }
+              return true;
             }
+            if (!value) throw new Error('Due date is required for non-recurring tasks');
             return true;
           },
         ],
         repeat_schedule: [
-          (value, { status }) => {
-            if (status === 'repeating' && !value) {
-              throw new Error('Repeat frequency is required for recurring tasks');
+          (value, { due_date: dueDate }) => {
+            // If the task has a due date, the repeat schedule is empty
+            if (dueDate) {
+              if (value) {
+                throw new Error('Non-recurring tasks cannot have a repeat schedule');
```

Comment:
```suggestion
                throw new Error('Recurring tasks cannot have a due date');
```
I might be confused but I'm not sure this message is correct

Reply:
Apologies, I have fixed the messages, I got confused writing them!